itou/utils/emails.py

```diff
@@ -69,3 +74,19 @@ def redact_part(part):
     else:
         redacted_domain = redact_part(domain_part)
     return f"{redacted_user}@{redacted_domain}"
+
+
+def check_sender_exists(func):
+    # some job application sender can be empty and were not checked when sending emails
+    # leading to 500 errors when refusing a job application
+    # not quite sure yet of the cause, but most probably because of as SET_NULL when deleting senders
+    # fixes: https://itou.sentry.io/issues/3401797706/events/89d13a2871bf453fa002e84127b297cf/?project=6164438
+    # see also: `job_application._clean_email_messages()`
```

Comment:
```suggestion
    # Most probably because of as SET_NULL when deleting senders
    # fixes: https://itou.sentry.io/issues/3401797706/events/89d13a2871bf453fa002e84127b297cf/?project=6164438
    # see also: `job_application._clean_email_messages()`
```